File: cs/DressDiscover/DressDiscover.Portable/src/DressDiscover/Api/Services/Worksheet/WorksheetQueryService.cs
namespace DressDiscover.Api.Services.Worksheet
{
    public interface WorksheetQueryService
    {
        System.Collections.Generic.List<string> GetWorksheetAccessionNumbers();
        DressDiscover.Api.Models.Worksheet.WorksheetDefinition GetWorksheetDefinition();
        DressDiscover.Api.Models.Worksheet.WorksheetState GetWorksheetState(string accessionNumber);
    }
}

File: ts/worksheet/src/dressdiscover/api/services/worksheet/worksheet_query_service.ts
import { WorksheetDefinition } from "../../models/worksheet/worksheet_definition";
import { WorksheetState } from "../../models/worksheet/worksheet_state";

export interface WorksheetQueryService {
    getWorksheetAccessionNumbersAsync(kwds: {error?: (errorKwds: {textStatus: string, errorThrown: any, [index: string]: any}) => any, success?: (returnValue: string[]) => void}): void;
    getWorksheetAccessionNumbersSync(): string[];

    getWorksheetDefinitionAsync(kwds: {error?: (errorKwds: {textStatus: string, errorThrown: any, [index: string]: any}) => any, success?: (returnValue: WorksheetDefinition) => void}): void;
    getWorksheetDefinitionSync(): WorksheetDefinition;

    getWorksheetStateAsync(kwds: {accessionNumber: string, error?: (errorKwds: {textStatus: string, errorThrown: any, [index: string]: any}) => any, success?: (returnValue: WorksheetState) => void}): void;
    getWorksheetStateSync(kwds: {accessionNumber: string}): WorksheetState;
}
